Framework/Core/Windows/Devices/Pwm/PwmController.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using Windows.Devices.Pwm.Provider;
+
+namespace Windows.Devices.Pwm
+{
+    public sealed class PwmController
+    {
+        private IPwmControllerProvider m_provider;
+        private double m_minFrequency;
+        private double m_maxFrequency;

```

Comment:
This isn't a performance optimization. It's insulation for Providers from common API violations (altering a property that's intended to be constant).  I've removed the caching anyway, but this does remove our ability to arbitrate intended behavior.